staging/src/k8s.io/kube-proxy/config/v1alpha1/types.go

```diff
@@ -85,6 +85,18 @@ type KubeProxyConntrackConfiguration struct {
 	// in CLOSE_WAIT state will remain in the conntrack
 	// table. (e.g. '60s'). Must be greater than 0 to set.
 	TCPCloseWaitTimeout *metav1.Duration `json:"tcpCloseWaitTimeout"`
+	// tcpBeLiberal, if true, kube-proxy will configure conntrack
+	// to run in liberal mode for TCP connections and packets with
+	// out-of-window sequence numbers won't be marked INVALID.
+	TCPBeLiberal *bool `json:"tcpBeLiberal"`
+	// udpTimeout is how long an idle UDP conntrack entry in
+	// UNREPLIED state will remain in the conntrack table
+	// (e.g. '30s'). Must be greater than 0 to set.
+	UDPTimeout metav1.Duration `json:"udpTimeout"`
+	// udpStreamTimeout is how long an idle UDP conntrack entry in
+	// ASSURED state will remain in the conntrack table
+	// (e.g. '300s'). Must be greater than 0 to set.
+	UDPStreamTimeout metav1.Duration `json:"udpStreamTimeout"`
```

Comment:
pointer

Reply:
will update that,
Also, just curious what will be the difference between `metav1.Duration` and `*metav1.Duration` here ?
I see [8 fields with](https://grep.app/search?q=%20metav1.Duration%24&regexp=true&filter[repo][0]=kubernetes/kubernetes&filter[path][0]=pkg/proxy/apis/config/&filter[lang][0]=Go) type `metav1.Duration` and [only 2 with type](https://grep.app/search?q=%5C%2Ametav1.Duration%24&regexp=true&filter[repo][0]=kubernetes/kubernetes&filter[path][0]=pkg/proxy/apis/config/&filter[lang][0]=Go) `*metav1.Duration`

Reply:
https://github.com/kubernetes/kubernetes/issues/119325 pointers default to nil so you can differentiate between the value is set and the default, for others you have to deal with golang defaults and you don't know if the value was set

Reply:
what about the serialization problem explained in https://github.com/kubernetes/kubernetes/issues/119325?

the problem the `0` vs `0s`, let me as @liggitt since he is more familiar with this problem but I feel that pointers are safer and easier to think about, however, better wait for experts opinions like Jordan or @thockin

Reply:
We can't use any `*metav1.Duration` directly in pflag unless there is a default value for it, If we want to use `*metav1.Duration` then we would need to sync the command line flag manually as we did for `TCPBeLiberal` here

```go
fs.DurationVar(&o.config.Conntrack.UDPTimeout.Duration, ....)
```

This will result in sigsegv if the value for `o.config.Conntrack.UDPTimeout` is nil. Also If I remove defaulting for either `TCPCloseWaitTimeout` or `TCPEstablishedTimeout` proxy panics during boot. 

Attempted with `*metav1.Duration` for  `UDPTimeout` and `UDPStreamTimeout ` here and the unit test also fails with sigsegv
 https://github.com/kubernetes/kubernetes/pull/120469

Reply:
we won't need to separately sync the command line flag and config if we implement something like `utilflag.MetaV1Duration` and 
 wrap the config variable in it like we do for bind-address https://github.com/kubernetes/kubernetes/blob/9f3cfbbcdbc8971dddfd7a6ba2c8eff10b0c9c82/cmd/kube-proxy/app/server.go#L166

Reply:
This PR moved from a value to a pointer https://github.com/kubernetes/kubernetes/pull/55261

Reply:
> why do we have these types in staging ?????

kubeproxyconfigv1alpha1 is picked from k8s.io/kube-proxy which I think is synced from staging.

https://github.com/kubernetes/kubernetes/blob/58ce7342230a943b637ff3d129eb3774d5a68cbb/pkg/proxy/apis/config/v1alpha1/register.go#L19-L22

Reply:
Yeah, sorry, I realized later and I forget to update the comment, thanks for doing the exercise